helix-lsp/src/lib.rs

```diff
@@ -205,6 +211,13 @@ pub mod util {
                 let line_start = doc.char_to_utf16_cu(doc.line_to_char(line));
                 let col = doc.char_to_utf16_cu(pos) - line_start;
 
+                lsp::Position::new(line as u32, col as u32)
+            }
+            OffsetEncoding::Utf32 => {
+                let line = doc.char_to_line(pos);
+                let line_start = doc.line_to_byte(line);
```

Comment:
For UTF-32 shouldn't this be `doc.line_to_char(line)` instead?

Reply:
ah good catch! I forgot to change one occurrence here thanks :)